frontend/src/lib/types.ts

```diff
@@ -0,0 +1,7 @@
+export interface Activity {
```

Comment:
Reduced the number of redundant fields - make invalid states impossible to represent (e.g. having bought -5 items, which was possible in last version).